packages/record-data/addon/-private/record-data.ts

```diff
@@ -293,7 +293,7 @@ export default class RecordDataDefault implements RelationshipRecordData {
     }
 
     this._isNew = false;
-    let newCanonicalAttributes: AttributesHash | null = null;
+    let newCanonicalAttributes: AttributesHash | undefined;
```

Comment:
Don't we want explicit null instead of setting to undefined?

Reply:
no, `attributes: null` is not a valid json:api payload, either the hash isn't present at all or its a hash.